partiql-planner/src/main/kotlin/org/partiql/planner/internal/helpers/ToBinder.kt

```diff
@@ -38,37 +38,22 @@ internal fun Expr.toBinder(index: () -> Int): Identifier = when (this) {
  * @param index
  * @return
  */
-internal fun Expr.toBinder(index: Int): Identifier = toBinder { index }
+internal fun Expr.toBinder(index: Int): Part = toBinder { index }
 
-private fun String.toBinder(): Identifier =
+private fun String.toBinder(): Part =
     // Every binder preserves case
-    identifier(this@toBinder, true)
+    identifierPart(this@toBinder, true)
 
-private fun IdentifierChain.toBinder(): Identifier {
-    if (next == null) return root.symbol.toBinder()
-    var cur = next
-    var prev = cur
-    while (cur != null) {
-        prev = cur
-        cur = cur.next
-    }
-    return prev!!.root.symbol.toBinder()
-}
```

Comment:
(self-review) example of the complicated logic to extract the "base" of a potentially qualified identifier. This is much simpler with the new modeling -- `this.base.toBinder()`